llvm/lib/Transforms/IPO/MergeFunctions.cpp

```diff
@@ -740,6 +740,10 @@ void MergeFunctions::writeThunk(Function *F, Function *G) {
   } else {
     NewG->copyAttributesFrom(G);
     NewG->takeName(G);
+    // Ensure CFI type metadata is propagated to the new function.
+    if (MDNode *TypeMD = G->getMetadata("type")) {
+      NewG->setMetadata("type", TypeMD);
+    }
```

Comment:
Do we need to copy other metadata as well, such as `kcfi_type`?

Reply:
I have created a helper function in case others identify additional metadata that should be copied over.